### PR TITLE
feat: Spotify agent chat integration with @-mentions and polling

### PR DESCRIPTION
**Pull request by brettbergin**
## Summary

Integrates the Spotify agent into the chat UI with full @-mention support and task result delivery.

### Changes

**Backend:**
- `@agent-slug` mention parsing in chat WebSocket handler (e.g. `@spotify what's playing?`)
- Mention context injection into LLM messages for targeted agent dispatch
- Updated system prompt with @-mention documentation
- Standardized `SpotifyAgent.execute()` result format (`summary` key)
- Fixed Spotify OAuth for headless Docker environments (`open_browser=False`, token cache check, clear auth URL error)
- Added `result` key fallback in worker summary extraction

**Frontend:**
- Polls for task results after dispatch (3s interval, 60s timeout)
- Full DB reload on poll to fix message ordering race condition
- `skipNextReloadRef` to prevent race on new conversation creation

### End-to-End Flow
```
User: "@spotify what's playing?"
  → WebSocket → LLM detects @spotify → dispatches task
  → LLM responds: "Checking Spotify..."
  → Worker executes SpotifyAgent → persists result to DB
  → Frontend polls → detects new message → renders result
```

### Files Modified
- `src/angie/api/routers/chat.py` — @-mention parsing and context injection
- `src/angie/agents/media/spotify.py` — Standardized results, headless OAuth
- `src/angie/queue/workers.py` — Robust summary extraction
- `frontend/src/app/chat/page.tsx` — Polling mechanism, message ordering fix
- `prompts/angie.md` — @-mention behavior docs

### CODE REVIEW

File: src/angie/agents/media/spotify.py [hunk 1]
```diff
@@ -124,18 +124,27 @@ def _build_client(self) -> Any:
...
+            client_id=os.environ.get("SPOTIFY_CLIENT_ID", ""),
+            client_secret=os.environ.get("SPOTIFY_CLIENT_SECRET", ""),
+            redirect_uri=os.environ.get("SPOTIFY_REDIRECT_URI", "http://127.0.0.1:8888/callback"),
+            scope="user-read-playback-state user-modify-playback-state user-read-currently-playing",
+            cache_path=cache_path,
+            open_browser=False,
         )
 
+        # Check if we have a cached token; if not, raise a clear error
+        token_info = auth_manager.cache_handler.get_cached_token()
```
[the remark this answers is not in the document]
Comment (brettbergin) [reply src/angie/agents/media/spotify.py]: ✅ Fixed. Token expiration check with refresh logic was already implemented — checks `is_token_expired()`, attempts `refresh_access_token()`, falls back to re-authorization if refresh fails. Already in place from a prior commit.

File: frontend/src/components/chat/MentionAutocomplete.tsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    ? items.filter((i) => i.slug.includes(query) || i.name.toLowerCase().includes(query))
+    : [];
+
+  useEffect(() => {
+    setSelectedIndex(0);
+  }, [query]);
+
+  const handleKeyDown = (e: React.KeyboardEvent) => {
+    if (!visible || filtered.length === 0) return false;
+
```
[the remark this answers is not in the document]
Comment (brettbergin) [reply frontend/src/components/chat/MentionAutocomplete.tsx]: ✅ Fixed. Added `import type React from 'react'` to MentionAutocomplete.tsx. See [ffb8503](https://github.com/brettbergin/angie/commit/ffb8503).

File: src/angie/api/routers/chat.py [hunk 3]
```diff
@@ -286,6 +383,8 @@ async def chat_ws(websocket: WebSocket, token: str, conversation_id: str | None
             response = {"content": reply, "role": "assistant"}
             if conversation_id:
                 response["conversation_id"] = conversation_id
+            if task_dispatched:
+                response["task_dispatched"] = True
             await websocket.send_text(json.dumps(response))
```
[the remark this answers is not in the document]
Comment (brettbergin) [reply src/angie/api/routers/chat.py]: ✅ Fixed. 8 unit tests added in `tests/unit/test_extract_mention.py` covering agent mentions, team mentions, invalid slugs, email edge case, case sensitivity, and positional variations. The `task_dispatched` flag now uses a simple `dispatch_flag` closure rather than pydantic-ai introspection. See [ffb8503](https://github.com/brettbergin/angie/commit/ffb8503).
